### PR TITLE
Adding PolicyInsights policyStates api version 2019-10-01.

### PR DESCRIPTION
**Pull request by sandipsh**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by weidongxu-microsoft — APPROVED** [review]

File: specification/policyinsights/resource-manager/readme.md [hunk 1]
```diff
@@ -56,6 +62,14 @@ directive:
       - $.paths["/{scope}/providers/Microsoft.PolicyInsights/policyStates/$metadata"].get.produces[0]
 
 ```
+### Tag: package-2019-10
+
+These settings apply only when `--tag=package-2019-10` is specified on the command line.
+
+``` yaml $(tag) == 'package-2019-10'
+input-file:
+- Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json
```
Comment (weidongxu-microsoft) [inline specification/policyinsights/resource-manager/readme.md]: Should it include all related json files (like policyEvents etc. as that is defined in package-2018-07), to compose a complete package?

**Comment by weidongxu-microsoft** [comment]
I am good with it (there is no API change). Please confirm on my last comment on whether you would like to put other json into package-2019-10. Then we could merge.

File: specification/policyinsights/resource-manager/readme.md (hunk 1 above)
Comment (sandipsh) [reply specification/policyinsights/resource-manager/readme.md]: Adding other json files to the package now.

**Comment by weidongxu-microsoft** [comment]
To ARM review. Commit changing the API is the last one https://github.com/Azure/azure-rest-api-specs/pull/7505/commits/964ec2957f79ce66ce1f0010e15d774904c753c7.

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyStates_QueryResourceGroupLevelPolicyAssignmentScope.json [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            "policyAssignmentId": "/subscriptions/fffedd8f-ffff-fffd-fffd-fffed2f84852/resourceGroups/myResourceGroup/providers/Microsoft.Authorization/policyAssignments/myPolicyAssignment",
+            "policyDefinitionId": "/subscriptions/fffedd8f-ffff-fffd-fffd-fffed2f84852/providers/Microsoft.Authorization/policyDefinitions/4a0425e4-97bf-4ad0-ab36-145b94083c60",
+            "effectiveParameters": null,
+            "isCompliant": false,
+            "subscriptionId": "fffedd8f-ffff-fffd-fffd-fffed2f84852",
+            "resourceType": "/Microsoft.Security/policies",
+            "resourceLocation": null,
+            "resourceGroup": "myResourceGroup",
+            "resourceTags": "tbd",
+            "policyAssignmentName": "myPolicyAssignment",
```
Comment (cemheren) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyStates_QueryResourceGroupLevelPolicyAssignmentScope.json]: Shouldn't these contain the new fields "complianceState" and "PolicyDefinitionGroupNames" ?

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyStates_GetMetadata.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "parameters": {
```
Comment (cemheren) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyStates_GetMetadata.json]: We retired the metadata API, after discussing with the ARM team. I think better to remove it from this version.

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyStates_SummarizeManagementGroupScope.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    "200": {
+      "body": {
+        "@odata.context": "https://management.azure.com/providers/Microsoft.Management/managementGroups/myManagementGroup/providers/Microsoft.PolicyInsights/policyStates/$metadata#summary",
+        "@odata.count": 1,
+        "value": [
+          {
+            "@odata.id": null,
+            "@odata.context": "https://management.azure.com/providers/Microsoft.Management/managementGroups/myManagementGroup/providers/Microsoft.PolicyInsights/policyStates/$metadata#summary/$entity",
+            "results": {
+              "queryResultsUri": "https://management.azure.com/providers/Microsoft.Management/managementGroups/myManagementGroup/providers/Microsoft.PolicyInsights/policyStates/latest/queryResults?api-version=2019-10-01&$from=2019-10-05 18:00:00Z&$to=2019-10-06 18:00:00Z&$filter=(PolicyDefinitionAction eq 'deny' or PolicyDefinitionAction eq 'audit') and IsCompliant eq false",
```
Comment (cemheren) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyStates_SummarizeManagementGroupScope.json]: These should have the new response format right?

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyStates_SummarizeResourceGroupLevelPolicyAssignmentScope.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+              "nonCompliantResources": 7,
+              "nonCompliantPolicies": 1
+            },
+            "policyAssignments": [
+              {
+                "policyAssignmentId": "/subscriptions/fffedd8f-ffff-fffd-fffd-fffed2f84852/resourcegroups/myResourceGroup/providers/microsoft.authorization/policyassignments/b7a1ca2596524e3ab19597f2",
+                "results": {
+                  "queryResultsUri": "https://management.azure.com/subscriptions/fffedd8f-ffff-fffd-fffd-fffed2f84852/resourceGroups/myResourceGroup/providers/Microsoft.Authorization/policyAssignments/b7a1ca2596524e3ab19597f2/providers/Microsoft.PolicyInsights/policyStates/latest/queryResults?api-version=2019-10-01&$from=2019-10-22 23:54:22Z&$to=2019-10-23 23:54:22Z&$filter=IsCompliant eq false and PolicyAssignmentId eq '/subscriptions/fffedd8f-ffff-fffd-fffd-fffed2f84852/resourcegroups/myResourceGroup/providers/microsoft.authorization/policyassignments/b7a1ca2596524e3ab19597f2'",
```
Comment (cemheren) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyStates_SummarizeResourceGroupLevelPolicyAssignmentScope.json]: All these need new format too

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyStates_QuerySubscriptionLevelPolicySetDefinitionScope.json [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            "resourceType": "/Microsoft.Web/sites",
+            "resourceLocation": "centralus",
+            "resourceGroup": "myResourceGroup",
+            "resourceTags": "tbd",
+            "policyAssignmentName": "3f3c4330183b4e218fe6fd29",
+            "policyAssignmentOwner": "tbd",
+            "policyAssignmentParameters": "{}",
+            "policyAssignmentScope": "/subscriptions/fffedd8f-ffff-fffd-fffd-fffed2f84852",
+            "policyDefinitionName": "24813039-7534-408a-9842-eb99f45721b1",
+            "policyDefinitionAction": "audit",
```
Comment (cemheren) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyStates_QuerySubscriptionLevelPolicySetDefinitionScope.json]: new columns

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json [hunk 7]
```diff
@@ -0,0 +1,1737 @@
...
+            "$ref": "#/parameters/fromParameter"
+          },
+          {
+            "$ref": "#/parameters/toParameter"
+          },
+          {
+            "$ref": "#/parameters/filterParameter"
+          },
+          {
+            "$ref": "#/parameters/applyParameter"
```
Comment (cemheren) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: what about expand for this

**Review by cemheren — CHANGES_REQUESTED** [review]

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json (hunk 7 above)
Comment (sandipsh) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: There's a separate path with expandParameter at resourceId scope that we currently support. I believe it's not supported at other scopes.

---
In reply to: [335637526](https://github.com/Azure/azure-rest-api-specs/pull/7505#discussion_r335637526) [](ancestors = 335637526)

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyStates_QuerySubscriptionLevelPolicySetDefinitionScope.json (hunk 6 above)
Comment (sandipsh) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyStates_QuerySubscriptionLevelPolicySetDefinitionScope.json]: Added.

**Review by cemheren — APPROVED** [review]

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json [hunk 8]
```diff
@@ -0,0 +1,1695 @@
...
+          "items": {
+            "$ref": "#/definitions/ComplianceDetail"
+          }          
+        }
+      }
+    },
+    "ComplianceDetail": {
+      "description": "The compliance state rollup.",
+      "properties": {
+        "complianceState": {
```
Comment (majastrz) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: >complianceState [](start = 9, length = 15)

Can this be modeled as an enum? (with modelAsString = true)

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json [hunk 9]
```diff
@@ -0,0 +1,1695 @@
...
+          "type": "string"
+        },
+        "policyDefinitionReferenceId": {
+          "description": "Reference ID for the policy definition inside the policy set, if the policy assignment is for a policy set.",
+          "type": "string"
+        },
+        "complianceState": {
+          "description": "Compliance state of the resource.",
+          "type": "string"
+        },
```
Comment (majastrz) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json]: Looking at example, this also looks like an enum. Might be worth adding the valid values to make it easier for the user. 

If this compliance state and other one you're adding below is of the same type (same allowed values), then you should probably declare an enum type and use it in all the places where it fits.

**Review by majastrz — APPROVED** [review]
Looks good from ARM side.

**Pull request merged**